### PR TITLE
A Comically Large amount of Quality of Life Improvements

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About the Pull Request

This PR adds a button for ghosts to become an SCP, so you don't have to navigate the ghostly and abandoned verb tab. Also, it generalizes the character preferences, it replaces most regular stuff with tgui inputs, and all that. It also gets rid of the 'derelict ship' spawners to get rid of confusion.

## Why It's Good For The Game

It's cooler, it's better, it doesn't confuse newbies.

## Changelog

:cl:
add: Ghosts now have a button for becoming an SCP, the days of navigating the verb tabs is now OVER!
del: The derelict ship shit in character prefs is now gone.
qol: TGUI is now more standard when changing your character prefs.
admin: Most admin verbs using the standard BYOND input() are now using TGUI inputs for searching and easier application.
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
This is actually really good, but since the SCP refactor touches the code for becoming an SCP, this will have a merge conflict next tm. Im probably just gonna split the next uptime into two parts, where half is the scp code refactor and anything that doesn't conflict, and the other half will be everything else after the refactor has been merged and conflicts have been fixed.

File: code/_onclick/hud/ghost_screen_objects.dm [hunk 1]
```diff
@@ -11,6 +11,18 @@
 	if(istype(fh))
 		G.follow(fh)
 
+/obj/screen/ghost/become_scp
+	name = "Become SCP"
```
Comment (TheDarkElites) [inline code/_onclick/hud/ghost_screen_objects.dm]:
```suggestion
	name = "Become an SCP"
```

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

**Comment by 77seven77seven** [comment]
> This is actually really good, but since the SCP refactor touches the code for becoming an SCP, this will have a merge conflict next tm. Im probably just gonna split the next uptime into two parts, where half is the scp code refactor and anything that doesn't conflict, and the other half will be everything else after the refactor has been merged and conflicts have been fixed.

i see

File: code/modules/client/preference_setup/general/01_basic.dm [hunk 2]
```diff
@@ -58,7 +58,7 @@
 	var/datum/species/S = all_species[pref.species]
 
 	if(href_list["rename"])
-		var/raw_name = input(user, "Choose your character's name:", "Character Name")  as text|null
+		var/raw_name = tgui_input_text(user, "Choose your character's name:", "Character Name", "John Doe", MAX_NAME_LEN)
```
Comment (Tsurupeta) [inline code/modules/client/preference_setup/general/01_basic.dm]: Default name seems useless, since we have proc that disallows nulls

Comment (77seven77seven) [reply code/modules/client/preference_setup/general/01_basic.dm]: done

File: code/modules/submaps/submap_archetype.dm [hunk 3]
```diff
@@ -14,8 +14,8 @@
 	. = ..()
 
 // Generic ships to populate the list.
-/decl/submap_archetype/derelict
-	descriptor = "drifting wreck"
+// /decl/submap_archetype/derelict
```
Comment (TheDarkElites) [inline code/modules/submaps/submap_archetype.dm]: These can be deleted

File: code/modules/submaps/submap_archetype.dm [hunk 4]
```diff
@@ -14,8 +14,8 @@
...
+	// descriptor = "drifting wreck"
 
-/decl/submap_archetype/abandoned_ship
-	descriptor = "abandoned ship"
+// /decl/submap_archetype/abandoned_ship
```
Comment (TheDarkElites) [inline code/modules/submaps/submap_archetype.dm]: same here

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

**Review by TheDarkElites — DISMISSED** [review]
This is all good. Slotting in for test merge.

**Review by Tsurupeta — DISMISSED** [review]

File: code/modules/client/preference_setup/background/02_language.dm [hunk 5]
```diff
@@ -47,7 +47,7 @@
 		if(!LAZYLEN(available_languages))
 			alert(user, "There are no additional languages available to select.")
 		else
-			var/new_lang = input(user, "Select an additional language", "Character Generation", null) as null|anything in available_languages
+			var/new_lang = tgui_input_list(user, "Select an additional lnaguage.", "Character Generation", available_languages)
```
Comment (EgorDinamit) [inline code/modules/client/preference_setup/background/02_language.dm]:
```suggestion
			var/new_lang = tgui_input_list(user, "Select an additional language.", "Character Generation", available_languages)
```

**Review by EgorDinamit — APPROVED** [review]
No other issues I can notice; PR was previously approved and test-merged, let's get this merged.

**Comment by 77seven77seven** [comment]
> No other issues I can notice; PR was previously approved and test-merged, let's get this merged.

YIPPIE

**Pull request merged**